packages/import-handler/src/index.ts

```diff
@@ -1,41 +1,65 @@
-import {
-  EventFunction,
-  CloudFunctionsContext,
-} from '@google-cloud/functions-framework/build/src/functions'
 import { Storage } from '@google-cloud/storage'
-import { importCsv, UrlHandler } from './csv'
+import { importCsv } from './csv'
 import * as path from 'path'
-import { importMatterHistory } from './matterHistory'
+import { importMatterArchive, importMatterHistoryCsv } from './matterHistory'
 import { Stream } from 'node:stream'
 import { v4 as uuid } from 'uuid'
-import { CONTENT_FETCH_URL, createCloudTask, EMAIL_USER_URL } from './task'
+import { CONTENT_FETCH_URL, createCloudTask, emailUserUrl } from './task'
 
+import axios from 'axios'
 import { promisify } from 'util'
 import * as jwt from 'jsonwebtoken'
+import { Readability } from '@omnivore/readability'
+
+import * as Sentry from '@sentry/serverless'
+
+Sentry.GCPFunction.init({
+  dsn: process.env.SENTRY_DSN,
+  tracesSampleRate: 0,
+})
 
 const signToken = promisify(jwt.sign)
 
 const storage = new Storage()
 
-interface StorageEventData {
-  bucket: string
+const CONTENT_TYPES = ['text/csv', 'application/zip']
+
+export type UrlHandler = (ctx: ImportContext, url: URL) => Promise<void>
```

Comment:
Sorry, I don't quite understand this part. Could you help me to understand the need to define the type here and the param `ctx` containing `urlHandler` itself?

Reply:
Oh yeah, so we have a few different types of import we can do: a CSV import or two types of Matter file import. So we have one function that figures out which type to use, and that returns the 'handler' function that should be used. 

Mostly this makes it a bit easier to test, i can decouple the detection of the file type from the execution.